src/clock/mumemoment.cpp

```diff
@@ -146,133 +130,175 @@ MumeTimeEnum MumeMoment::toTimeOfDay() const
+           8-15: under the horizon*/
+    const int dir = std::clamp((now - rise) * 16 / MUME_MINUTES_PER_DAY, 0, 15);
+    if (dir < 8) {
+        static const constexpr int EAST = static_cast<int>(MumeMoonPositionEnum::EAST);
+        static const constexpr int MIN = static_cast<int>(MumeMoonPositionEnum::INVISIBLE);
+        static const constexpr int MAX = static_cast<int>(MumeMoonPositionEnum::WEST);
+        return static_cast<MumeMoonPositionEnum>(std::clamp(EAST + (dir + 1) / 2, MIN, MAX));
+    }
+    return MumeMoonPositionEnum::INVISIBLE;
 }
 
-int MumeMoment::moonSet() const
+int MumeMoment::moonLevel() const
 {
-    return (moonCycle() + 18) % MUME_DAYS_PER_MOON_CYCLE;
+    // Illumination level seems to be between 0-12
+    // 0 is new moon, 12 is full moon
+    // Levels > 4 make the moon show up in the prompt as ')'
+    int level = (moonZenithMinutes() + MUME_MINUTES_PER_HOUR / 2) / MUME_MINUTES_PER_HOUR;
+    return abs(12 - level);
 }
 
-int MumeMoment::moonPosition() const
+MumeMoonPhaseEnum MumeMoment::moonPhase() const
 {
-    // Position of moon in the sky out of 24
-    // Below horizon is position >= 12
-    return (hour - moonRise() + MUME_HOURS_PER_MOON_POSITION) % MUME_HOURS_PER_MOON_POSITION;
+    // Moon starts off as full phase at the start epoch
+    // Divide level by three for phase (new, 1/4, 1/2, 3/4, full)
+    const int phase = std::clamp(moonLevel() / 3, 0, 4);
+    static const constexpr MumeMoonPhaseEnum WAXING[] = {MumeMoonPhaseEnum::NEW_MOON,
+                                                         MumeMoonPhaseEnum::WAXING_CRESCENT,
+                                                         MumeMoonPhaseEnum::FIRST_QUARTER,
+                                                         MumeMoonPhaseEnum::WAXING_GIBBOUS,
+                                                         MumeMoonPhaseEnum::FULL_MOON};
+    static const constexpr MumeMoonPhaseEnum WANING[] = {MumeMoonPhaseEnum::NEW_MOON,
+                                                         MumeMoonPhaseEnum::WANING_CRESCENT,
+                                                         MumeMoonPhaseEnum::THIRD_QUARTER,
+                                                         MumeMoonPhaseEnum::WANING_GIBBOUS,
+                                                         MumeMoonPhaseEnum::FULL_MOON};
+    return isMoonWaxing() ? WAXING[phase] : WANING[phase];
 }
 
-int MumeMoment::moonLevel() const
+MumeMoonVisibilityEnum MumeMoment::moonVisibility() const
 {
-    // Illumination level seems to be between 0-12
-    // Levels > 4 make the moon show up in the prompt as ')'
-    const int dayOfCycle = moonCycle();
-    if (dayOfCycle <= 12)
-        return dayOfCycle; // Increasing illumination
-    else
-        return MUME_DAYS_PER_MOON_CYCLE - dayOfCycle; // Decreasing illumination
+    if (isMoonBelowHorizon() || moonPhase() == MumeMoonPhaseEnum::NEW_MOON)
+        return MumeMoonVisibilityEnum::INVISIBLE;
+
+    const auto isBright = isMoonBright();
```

Comment:
I personally would prefer just renaming the members to be a bit shorter over this aliasing idiom, but nbd either way

Reply:
Yeah, I'm a little tired of staring at this code and just want to get it out there to users. It's more important at this point that it works rather than it being perfectly maintainable. All it takes is Dain changing it once more... :-P